src/pyfmi/fmi3.pyx

```diff
@@ -675,6 +678,168 @@ cdef class FMUModelBase3(FMI_BASE.ModelBase):
+
+            causality --
+                The causality of the variables as an instance of pyfmi.fmi3.FMI3_Causality, int, or None.
+                Default: None (i.e all).
+
+            variability --
+                The variability of the variables as an instance of pyfmi.fmi3.FMI3_Variability, int, or None.
+                Default: None (i.e all).
+
+            only_start --
+                If only variables that has a start value should be returned.
+                Default: False
+
+            only_fixed --
+                If only variables that has a start value that is fixed should be returned.
+                Default: False
+
+            filter --
+                Filter the variables using a unix filename pattern
+                matching (filter="*der*"). Can also be a list of filters
+                See http://docs.python.org/2/library/fnmatch.html.
+                Default: None
+
+        Returns::
+
+            Dict with variable name as key and a pyfmi.fmi3.FMI3ModelVariable class as value.
+        """
+        return self._get_model_variables(
+            type = int(type) if type is not None else None,
+            include_alias = include_alias,
+            causality = int(causality) if causality is not None else None,
+            variability = int(variability) if variability is not None else None,
+            only_start = only_start,
+            only_fixed = only_fixed,
+            filter = filter
+        )
+
+    @enable_caching
+    def _get_model_variables(self,
+        type,
+        include_alias,
+        causality,
+        variability,
+        only_start,
+        only_fixed,
+        filter):
+        cdef FMIL3.fmi3_import_variable_t*        variable
+        cdef FMIL3.fmi3_import_variable_list_t*   variable_list
+        cdef FMIL.size_t                          variable_list_size
+        cdef FMIL3.fmi3_value_reference_t         value_ref
+        cdef FMIL3.fmi3_base_type_enu_t           data_type
+        cdef FMIL3.fmi3_variability_enu_t         data_variability
+        cdef FMIL3.fmi3_causality_enu_t           data_causality
+        cdef FMIL3.fmi3_initial_enu_t             data_initial
+
+        cdef int has_start = 0
+        # TODO: Can we rename the keyword arg 'filter' to variable_filter?
+        variable_filter = filter
+        # TODO: Can we rename the keyword arg 'type' to variable_type?
+        variable_type = type
+        cdef list filter_list, variable_return_list = []
+        variable_dict = {}
+
+        variable_list      = FMIL3.fmi3_import_get_variable_list(self._fmu, 0)
+        variable_list_size = FMIL3.fmi3_import_get_variable_list_size(variable_list)
+
+        if variable_filter:
+            filter_list = self._convert_filter(variable_filter)
+
+        user_specified_type        = isinstance(variable_type, int)
+        user_specified_variability = isinstance(variability, int)
+        user_specified_causality   = isinstance(causality, int)
```

Comment:
@PeterMeisrimelModelon  d8aaeea

Reply:
Based on the inputs we know it is `int` or `None`, I wonder if it was faster to instead check  to do `user_specified_type = variable_type is not None`. 

Feel free to test/ignore/fix/resolve.